File: addaprime.php
<?php

function secret($number) {
	return $number;
}

function all_primes() {
	global $argv;
	$input = intval($argv[1]);
	if( $input < 3 ) die("Please input an integer larger than 2.");
	$primes = array();
	for( $i = 2; $i <= $input; $i = gmp_nextprime($i)) {
		array_push($primes, gmp_strval($i));
	}
	return $primes;
}

function secret_together($x, $y) {
	return secret($x + $y);
}

function secret_apart($x, $y) {
	return secret($x) + secret($y);
}

File: addaprime.rb
require 'prime'
input = ARGV[0].to_i

def secret(number)
  number
end

def all_primes(input)
  Prime.each(input).to_a
end

def secret_together(x,y)
  secret(x+y)
end

def secret_apart(x,y)
  secret(x) + secret(y)
end

def all_combinations(input)
  combinations = Array.new
  all_primes(input).each do |x|
    all_primes(input).each do |y|
      combinations.push([x,y])
    end
  end
  combinations
end

def additive?(combinations)
  combinations.each do |x,y|
    return false unless secret_together(x,y) == secret_apart(x,y)
  end
  true
end

c = all_combinations(input)
if additive?(c)
  puts "Congratulations. The secret() function is indeed additive."
else
  puts "Sorry. The secret() function is not additive."
end

